pkg/build/util/secrets.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"encoding/json"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/credentialprovider"
+	"github.com/golang/glog"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// DockerConfigFromSecret parse the Secret 'data' and return the DockerConfig
+// which contains representation of the '.dockercfg'.
+// FIXME: This was copied from upstream because the upstream function was not
+//        public.
+func DockerConfigFromSecret(secret *kapi.Secret) (cfg *credentialprovider.DockerConfig) {
+	if secret.Type != kapi.SecretTypeDockercfg {
+		return nil
+	}
+	if err := json.Unmarshal(secret.Data[kapi.DockerConfigKey], cfg); err != nil {
+		glog.V(2).Infof("Unable to parse the .dockercfg content in secret %s: %v", secret.Name, err)
+		return nil
+	}
+	return cfg
+}
+
+// GetImageDockerConfigEntry retrieves the credentials from the DockerConfig
+// (if found) that allows pushing or pulling from the private Docker registry.
+func GetImageDockerConfigEntry(imageSpec string, cfg credentialprovider.DockerConfig) *credentialprovider.DockerConfigEntry {
+	image, err := imageapi.ParseDockerImageReference(imageSpec)
+	if err != nil {
+		glog.V(2).Infof("Unable to parse the Docker image '%s': %v", imageSpec, err)
+		return nil
+	}
+	if e, ok := cfg[image.Registry]; ok {
```

Comment:
@ncdc the value of image.Registry here compatible with the .dockercfg entry name format for registries? IOW. does it return `https://index.docker.io/v1/` for DockerHub ?